core/src/modules/optionHelper.ts

```diff
@@ -323,6 +342,57 @@ export namespace OptionHelper {
     return propertyOptionOptions;
   }
 
+  export async function getOptionOptions(
+    instance: ModelWithOptions,
+    options: SimpleOptions
+  ): Promise<Record<string, string[]>> {
+    if (instance instanceof Property) {
+      const pluginOptions = await instance.pluginOptions(options);
+      return optionOptionsFromArray(pluginOptions);
+    } else if (instance instanceof Schedule) {
+      const pluginOptions = await instance.pluginOptions();
+      return optionOptionsFromArray(pluginOptions);
+    } else if (instance instanceof Destination) {
+      const destinationConnectionOptions =
+        await instance.destinationConnectionOptions(options);
+      return optionOptionsFromObject(destinationConnectionOptions);
+    } else if (instance instanceof App) {
+      const appOptions = await instance.appOptions();
+      return optionOptionsFromObject(appOptions);
+    } else if (instance instanceof Source) {
+      const sourceConnectionOptions = await instance.sourceConnectionOptions(
+        options
+      );
+      return optionOptionsFromObject(sourceConnectionOptions);
+    }
+  }
```

Comment:
Now we've got 2 big case statements in this helper (this + validateOptions) that differ per model.  This sort of looks like we should be doing less in this helper and more per-model maybe?

Reply:
Yeah, that's a good point. These are _similar_ but they each have slight differences.

I refactored the `validateOptions` in dfb3edf / 5b82795 to be called from the model so we can get the data in the right format and then pass it to the common validation logic.